undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java

```diff
@@ -427,6 +427,12 @@ public synchronized void start(final StartContext startContext) throws StartExce
 
     @Override
     public synchronized void stop(final StopContext stopContext) {
+        // Remove the server activity
+        final ServerActivity activity = serverActivity.get();
+        if(activity != null) {
+            suspendController.get().unRegisterActivity(activity);
+        }
+        UndertowContainerProvider.removeContainer(module.getClassLoader());
```

Comment:
I think we'd want to actually call io.undertow.websockets.jsr.SecurityActions.removeContainer instead but this is protected:
https://github.com/undertow-io/undertow/blob/master/websockets-jsr/src/main/java/io/undertow/websockets/jsr/SecurityActions.java#L40

So do we change that from protected?  Or add the security manager handling to UndertowDeploymentInfoService?

Reply:
It shouldn't matter. An MSC thread calling stop shouldn't have anything but container classes on the call stack.

Reply:
Thanks @fl4via, added it to UndertowDeploymentInfoService